main.go

```diff
@@ -130,17 +132,51 @@ func main() {
 
 	}
 
+	// setup metrics via dropsonse
 	sender := metric_sender.NewMetricSender(dropsonde.AutowiredEmitter())
+	var metricsReporter *metrics.MetricsReporter
+	if c.EnableEnvelopeV1Metrics {
+		metricsReporter = initializeMetrics(sender, c, grlog.CreateLoggerWithSource(prefix, "metricsreporter"))
+	}
+
+	var promRegistry *mr.Registry
+	var promMetrics *metrics_prometheus.Metrics
+	if c.Prometheus.Enabled {
+		promRegistry = metrics_prometheus.NewMetricsRegistry(c.Prometheus)
+		promMetrics = metrics_prometheus.NewMetrics(promRegistry, c.PerRequestMetricsReporting, c.Prometheus.Meters)
+	}
+
+	// setup metrics via prometheus
+	var registryMetrics metrics.MultiRouteRegistryReporter
+
+	if metricsReporter != nil {
+		registryMetrics = append(registryMetrics, metricsReporter)
+	}
+	if promMetrics != nil {
+		registryMetrics = append(registryMetrics, promMetrics)
+	}
```

Comment:
You could do the initialization of dropsonde, or prometheus, or both in a separate function that then returns a registry reporter, either a "regular" one or the `MultiRouteRegistryReporter`.